src/main/java/org/diatliuk/bookstore/repository/book/BookRepository.java

```diff
@@ -1,9 +1,30 @@
 package org.diatliuk.bookstore.repository.book;
 
+import jakarta.validation.constraints.NotNull;
+import java.util.List;
+import java.util.Optional;
 import org.diatliuk.bookstore.model.Book;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.domain.Specification;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
+import org.springframework.data.jpa.repository.Query;
 
 public interface BookRepository extends JpaRepository<Book, Long>, JpaSpecificationExecutor<Book> {
+    List<Book> findAllByCategoriesId(Long categoryId);
 
+    @NotNull
```

Comment:
What is the purpose of this annotation on repository methods?

Reply:
Frankly, I was confused about it, since it was suggested by Idea. But I guess it was my bad and I didn't get the Idea suggestions correctly